### DISCUSSION TITLE
拉取docker镜像报错

### DISCUSSION THREAD

**Discussion by andlink01** [kkkgo/PaoPaoDNS#233] [问答和帮助] [answered]
docker: Error response from daemon: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io: Temporary failure in name resolution.
See 'docker run --help'.

部署了paopaodns+paopaogw，拉取docker镜像就出现报错，配置了daemon.json也不行

**Comment by kkkgo** [comment]
failure in name resolution.是没有DNS解析 。

**Reply by andlink01** [reply]
![IMG_2283](https://github.com/user-attachments/assets/9b54fbbf-4e4f-49c2-870a-432115674bbc)
测试是返回了fakeip

**Reply by kkkgo** [reply]
1、你nslookup的终端和docker终端是否是同一个。
2、使用`curl -v https://registry-1.docker.io/v2/`如果能正常输出内容，你需要自行检查docker环境配置问题。

**Reply by andlink01** [reply]
> 1、你nslookup的终端和docker终端是否是同一个。 2、使用`curl -v https://registry-1.docker.io/v2/`如果能正常输出内容，你需要自行检查docker环境配置问题。

在同一个局域网内多台设备测试均是下边报错，晚上我部署dns的时候还是可以正常pull,并且我还配置了/etc/docker/daemon.json添加了加速。并且我也测试删除这个直接拉都是一样的错误。
root@debian-dns:~# curl -v https://registry-1.docker.io/v2/
* Could not resolve host: registry-1.docker.io
* Closing connection 0
curl: (6) Could not resolve host: registry-1.docker.io
root@debian-dns:~# 
root@debian-dns:~#

**Reply by kkkgo** [reply]
很明显你curl执行的就是一样的报错，curl: (6) Could not resolve host: registry-1.docker.io，就是没有DNS解析。
你的系统可能没有正确配置DNS地址，或者配置了其他不可用的DNS地址（比如IPv6地址）

**Reply by andlink01** [reply]
> 很明显你curl执行的就是一样的报错，curl: (6) Could not resolve host: registry-1.docker.io，就是没有DNS解析。 你的系统可能没有正确配置DNS地址，或者配置了其他不可用的DNS地址（比如IPv6地址）

dns肯定是有的，手工配置interfaces指定了的，并且还有resolvconf自动更新resolve.conf
我先试试关闭IPv6

**Reply by andlink01** [reply, marked as answer]
> 很明显你curl执行的就是一样的报错，curl: (6) Could not resolve host: registry-1.docker.io，就是没有DNS解析。 你的系统可能没有正确配置DNS地址，或者配置了其他不可用的DNS地址（比如IPv6地址）

找到问题了，部署PaoDNS的机器上运行了systemd-resolved，这个程序也会监听53端口，最开始我关闭了IPv6报错提示变了，但是依旧还是不能用。后来重启paodns的Docker发现提示端口被占用
